client/src/components/Schedule/useScheduleSettings.ts

```diff
@@ -0,0 +1,89 @@
+import { useMemo } from 'react';
+import { useLocalStorage } from 'react-use';
+import { DEFAULT_COLORS, DEFAULT_VIEW_MODE } from 'components/Schedule/constants';
+import { ViewMode } from './constants';
+
+enum LocalStorage {
+  ViewMode = 'scheduleViewMode',
+  Timezone = 'scheduleTimezone',
+  LimitForDoneTasks = 'scheduleLimitForDoneTask',
+  IsSplittedByWeek = 'scheduleIsSplitedByWeek',
+  ArePassedEventsHidden = 'scheduleArePassedEventsHidden',
+  AreDoneTasksHidden = 'scheduleAreDoneTasksHidden',
+  TagColors = 'scheduleTagColors',
+  ColumnsHidden = 'scheduleColumnsHidden',
+  EventTypesHidden = 'scheduleEventTypesHidden',
+}
+
+export interface ScheduleSettings {
+  viewMode: ViewMode;
+  setViewMode: (value: ViewMode) => void;
+  timezone: string;
+  setTimezone: (value: string) => void;
+  limitForDoneTask: number;
+  setLimitForDoneTask: (value: number) => void;
+  isSplittedByWeek: boolean;
+  setIsSplittedByWeek: (value: boolean) => void;
+  arePassedEventsHidden: boolean;
+  setArePassedEventsHidden: (value: boolean) => void;
+  areDoneTasksHidden: boolean;
+  setAreDoneTasksHidden: (value: boolean) => void;
+  tagColors: Record<string, string>;
+  setTagColors: (value: Record<string, string>) => void;
+  columnsHidden: string[];
+  setColumnsHidden: (value: string[]) => void;
+  eventTypesHidden: string[];
+  setEventTypesHidden: (value: string[]) => void;
+}
+
+const useScheduleSettings = (): ScheduleSettings => {
```

Comment:
also wondering about the approach: seems that majority of our code uses named exports and less defaults. wondering why do you prefer this approach?

Reply:
I export everything what we need outside in the `client/src/components/Schedule/index.ts`.
So we can import this hook as named one:
```
import { useScheduleSettings } from 'components/Schedule';
```